feature_engine/encoding/count_frequency.py

```diff
@@ -145,15 +148,18 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
         self._get_feature_names_in(X)
 
         self.encoder_dict_ = {}
+        dct_init = defaultdict(lambda: 0) if self.errors == "zeros" else {}
 
         # learn encoding maps
         for var in self.variables_:
             if self.encoding_method == "count":
-                self.encoder_dict_[var] = X[var].value_counts().to_dict()
+                self.encoder_dict_[var] = X[var].value_counts().to_dict(dct_init)
```

Comment:
what does this do?

Reply:
It starts from a base dictionary which gets updated with the counts.